### PR TITLE
Complain about HEP 6 violations.

### PR DESCRIPTION
**Pull request by tdsmith**
The shebang check may be too stringent; these may be small scripts that
don't touch the network and installing Python for them could be
overkill. Or maybe not!

### CODE REVIEW

File: Library/Homebrew/extend/os/mac/formula_cellar_checks.rb [hunk 1]
```diff
@@ -59,6 +59,33 @@ def check_python_framework_links(lib)
...
   end
 
+  def check_python_virtualenv
+    framework = formula.libexec/".Python"
+    return unless framework.symlink?
+    return unless framework.realpath.to_s.start_with?("/System")
+    <<-EOS.undent
+      virtualenv created against system Python
+      This formula created a virtualenv using system Python.
+      Please add `depends_on :python` to the formula.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/mac/formula_cellar_checks.rb]: `depends_on :python` may use the system Python, though, if there's no Homebrew one installed.

**Comment by tdsmith** [comment]
Wait, that last commit was dumb.

File: Library/Homebrew/extend/os/mac/formula_cellar_checks.rb [hunk 2]
```diff
@@ -59,6 +59,35 @@ def check_python_framework_links(lib)
...
+    EOS
+  end
+
+  def check_python_shebangs
+    return unless formula.bin.directory?
+    return if formula_requires_python?(formula) && default_python_is_system_python?
+    system_python_shebangs = formula.bin.children.select do |bin|
+      (bin.open { |f| f.read(32) }).start_with?("#!/usr/bin/python")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/mac/formula_cellar_checks.rb]: Why 32? If it's the string length can you make it a variable and get the length dynamically that way?

File: Library/Homebrew/extend/os/mac/formula_cellar_checks.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/mac/formula_cellar_checks.rb]: Again, still not quite sure I understand this message given the system Python behaviour.

File: Library/Homebrew/extend/os/mac/formula_cellar_checks.rb [hunk 3]
```diff
@@ -59,6 +59,35 @@ def check_python_framework_links(lib)
...
+    end
+    return if system_python_shebangs.empty?
+
+    <<-EOS.undent
+      python scripts run with system python
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/mac/formula_cellar_checks.rb]: Again, still not quite sure I understand this message given the system Python behaviour.

File: Library/Homebrew/extend/os/mac/formula_cellar_checks.rb (hunk 1 above)
Comment (tdsmith) [inline Library/Homebrew/extend/os/mac/formula_cellar_checks.rb]: The alert only fires if `depends_on :python` was not specified.

File: Library/Homebrew/extend/os/mac/formula_cellar_checks.rb (hunk 2 above)
Comment (tdsmith) [inline Library/Homebrew/extend/os/mac/formula_cellar_checks.rb]: Arbitrary short length. Sure, doesn't have to be magical.

File: Library/Homebrew/extend/os/mac/formula_cellar_checks.rb (hunk 1 above)
Comment (tdsmith) [inline Library/Homebrew/extend/os/mac/formula_cellar_checks.rb]: — The rationale is that `depends_on :python` will never fall back on system Python on the test bot, which is all I care about detecting.

**Comment by tdsmith** [comment]
Can you comment on the message you'd prefer to see and when this messaging will be confusing?

File: Library/Homebrew/extend/os/mac/formula_cellar_checks.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/mac/formula_cellar_checks.rb]: By "on the test bot" you mean "when building bottles", right? I'm not 100% sure if that happens if e.g. `bottle :disabled` or `bottle :unused` is set; may be worth testing.

**Comment by MikeMcQuaid** [comment]
> Can you comment on the message you'd prefer to see and when this messaging will be confusing?

I think it may be worth filtering and/or explicitly referencing in both messaging and code whether the formula was built as a bottle and that's the failure case. Arguably that's a good reason to perhaps punt this check to `brew bottle` too.

**Comment by MikeMcQuaid** [comment]
> these may be small scripts that don't touch the network and installing Python for them could be overkill

Given this assumes things communicating over the network and I'm unconvinced that's the majority case I'm probably 👎 on having an automated check for this. False positives are fine, false negatives are not.

**Comment by MikeMcQuaid** [comment]
Let's pass on this for now but thanks for the work @tdsmith.

**Pull request closed without merging**